compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +289,118 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
+    AstNode* ret = nullptr;
+    std::string configUsed;
+    auto configs = parsing::configParams(this->context());
+    if (auto var = ast->toVariable()) {
+      if (var->isConfig()) {
+        // inspect pathVec
+        std::string possibleModule;
+        if (pathVec.size() > 1) {
+          for (size_t i = 1; i < pathVec.size(); i++) {
+            possibleModule += pathVec[i].first.str();
+            possibleModule += ".";
+          }
+        } else if (pathVec.size() == 1) {
+          possibleModule = pathVec[0].first.str();
+          possibleModule += ".";
+        }
+
+        // for config vars, check if they were set from the command line
+        for (auto node : configs) {
+          // TODO: How does this work with module prefixes? What if there is a newSymbolPath value?
+          if ((node.first == var->name().str() && var->visibility() != Decl::PRIVATE)
+              || node.first == possibleModule + var->name().str()) {
+            // found a config that was set via cmd line: update the node
+            // handle deprecations
+            if (auto attribs = var->attributes()) {
+              if (attribs->isDeprecated()) {
+                // TODO: Need proper message handling here
+                std::string msg = "'" + var->name().str() + "' was set via a compiler flag";
+                std::cerr << "warning: " + attribs->deprecationMessage().str() << std::endl;
+                std::cerr << "note: " + msg << std::endl;
+              }
+            }
+            auto parser = parsing::Parser::build(context());
+            parsing::Parser* p = parser.get();
+            std::string inputText;
+            // for types, we can't specify a module name when building the input string
+            // and it's important for the parser to see that it's a type and not just a var assignment
+            if (var->kind() == uast::Variable::TYPE) {
+              inputText = "type " + var->name().str() + "=" + node.second +";";
+            } else {
+              inputText = (!node.second.empty()) ? node.first + "=" + node.second +";" : node.first + "=true;";
+            }
+            // TODO: how to handle nested module configs e.g., -sFoo.Baz.bar=10
+            auto parseResult = p->parseString("CompilationConfigs", inputText.c_str());
+            assert(!parseResult.numErrors());
+            auto mod = parseResult.singleModule();
+            assert(mod);
+            if (mod->stmt(0)->isOpCall()) {
+              assert(mod->stmt(0)->toOpCall()->isBinaryOp());
+              ret = mod->children_[0]->children_[1].release();
+            } else if (mod->stmt(0)->isVariable()) {
+              ret = mod->children_[0]->children_[0].release();
+            } else {
+              assert(false && "should only be an assignment or type initializer");
+            }
+            // TODO: How to handle locations? The column numbers at least should be updated, no?
```

Comment:
Right, the column numbers won't match. Since we generated the `var bla = bla` code to parse, we could just add spacing to get the column numbers to match. (e.g. `var bla =\n<variable amount of space>bla;`. Or we could decide that line numbers / column numbers don't make sense for command line stuff (since the location is just `<command line>`. That's the direction I'd suggest for now.